### ISSUE TITLE
selector is invoked for all of my dispatched actions

### ISSUE THREAD

**Issue by alexbollbach** [reactjs/react-redux#794]
I'm not sure if this is the best place to post usage questions.  But I've tried Stack Overflow, Discord channels, read for many hours, and rewrote most of my app but am still stuck.  

I have a React app that does some simple recording.  I have a Component `Recorder` which connects to my redux store like this:

    export default connect(
    	state => ({ 
    		recordings: state.recordings,
    		recordingSelector: selectRecordingBufferWithID(this.recordingID) 
    	}), 
    	
    	dispatch =>
    		bindActionCreators({ 
    				startNewRecordingAction, 
    				stopNewRecordingAction 
    			},
    			dispatch
    		)
    )(SampleRecorder);

The problem I'm having is that `selectRecordingBufferWithID` in my redux code is firing too often.  Part of my reducer code looks like this:

    function samplesReducer(state = [], action) {
    	switch (action.type) {
    		case MORE_SAMPLES:
    			return [...action.samples];
    		default:
    			return state
    	}	
    }
    
    function recordingsReducer(state = [], action) {
    	switch (action.type) {
    		case NEW_RECORDING:
    			return newRecording(state, action.recordingID);
    		case STOP_RECORDING:
    			return stopRecording(state, action.recordingID);
    		default:
    			return state
    
    	}
    }
    
    const rootReducer = combineReducers({
    	samplesReducer,
    	recordingsReducer
    })

    const store = createStore(rootReducer);
    export { store };

So, while I want `selectRecordingBufferWithID` to be utilized only when a `START/STOP_RECORDING` action occurs, it is called for each time `MORE_SAMPLES` is called.

-----

My understanding of `react-redux` is that the selector is part of the `mapStateToProps` function that the `connect` function accepts.  And somehow, `connect` cause my component to render and for its props to be updated with the mapped state from the redux store.  the `selectRecordingBufferWithID` selector will also be called each time this happens so I can do a refined getter into the store.  

So to summarize, my `recordingSelector` is firing more often than I expect.  My only theory is that my reducers are somehow mutating the state of `state.recordings` each time it tries to reduce `state.samples` which makes `react-redux` render my component with it mapped to `state.recording`.

But otherwise, I'm stuck.

**Comment by timdorr** [comment]
Sorry, but this is a bug tracker, not a support system. There isn't anywhere near as large an audience watching this issues as Stack Overflow or Discord. Make sure you're including sufficient info for what your issue is and that you're not making it too specific to the functionality of your app. That's the best way to get broad appeal for answering your question.
